app/lib/package/upload_signer_service.dart

```diff
@@ -28,6 +28,21 @@ UploadSignerService get uploadSigner =>
 void registerUploadSigner(UploadSignerService uploadSigner) =>
     ss.register(#_url_signer, uploadSigner);
 
+/// Creates an upload signer based on the current environment.
+Future<UploadSignerService> createUploadSigner(http.Client authClient) async {
+  if (envConfig.isRunningLocally) {
+    return ServiceAccountBasedUploadSigner();
```

Comment:
Can we make `ServiceAccountBasedUploadSigner` internal or `@visibleForTestingOnly`? Similar for `IamBasedUploadSigner`

Reply:
Done.